crypto/x509/v3_purp.c

```diff
@@ -469,11 +491,11 @@ static void x509v3_cache_extensions(X509 *x)
     x->akid = X509_get_ext_d2i(x, NID_authority_key_identifier, NULL, NULL);
     /* Does subject name match issuer ? */
     if (!X509_NAME_cmp(X509_get_subject_name(x), X509_get_issuer_name(x))) {
-        x->ex_flags |= EXFLAG_SI;
-        /* If SKID matches AKID also indicate self signed */
-        if (X509_check_akid(x, x->akid) == X509_V_OK &&
-            !ku_reject(x, KU_KEY_CERT_SIGN))
-            x->ex_flags |= EXFLAG_SS;
+        x->ex_flags |= EXFLAG_SI; /* cert is self-issued */
+        if (X509_check_akid(x, x->akid) == X509_V_OK /* SKID matches AKID */
+            && /* .. and the signature alg matches the PUBKEY alg: */
+            check_sig_alg_match(X509_get0_pubkey(x), x) == X509_V_OK)
```

Comment:
the indentation is wrong, we want the logical expressions in if / while indented 4 spaces.
and please place the comment on a extra line, also 4-space indented.
In case you want ask where that style comes from please look at the maling list here,
https://www.mail-archive.com/openssl-project@openssl.org/msg00164.html

and several discussions in the monts before, with topic "Style question" and
"Style guide update".

Reply:
Changed.